edgePy/DGEList.py

```diff
@@ -292,17 +292,40 @@ def rpkm(
         # self = self.cpm(log=log, prior_count=prior_count)
 
     def tpm(
-        self, transcripts: Mapping, log: bool = False, prior_count: float = PRIOR_COUNT
-    ) -> None:
-        """Return the DGEList normalized to reads per kilobase of transcript
-        length.
+        self,
+        gene_lengths: np.ndarray,
```

Comment:
We need an assertion (and raise exception) if this is not a np.array and the shape of this array is not 1D.

Do something like:

```python
if not isinstance(gene_lengths, np.ndarray):
    raise TypeError('...')
if gene_lengths.shape() != (1, 0):
    raise ValueError('...')
```

Do the same for `mean_fragment_lengths`.

Reply:
We should merge this branch with the af_rpkm branch to take advantage of the gene_lengths objets available there.